### PR TITLE
FEATURE: Mint

### PR DESCRIPTION
**Pull request by jmysliv**
* Added parsing `.env` file
* Added asking for media library permission for native
* Added Mint Form (changed designs a little bit)
* Refactored `Button` component so it can be reusable
* Refactored `EditProfile` to match the design of the mint form
* Added `useMintNFT` hook that handles NFT minting
* Added `retryIndexing`

DEMO:

https://user-images.githubusercontent.com/48885911/189160313-b4a56f9b-05bd-4bc1-9b39-ece64e1655f8.mp4

Screenshot 2022-09-08 at 12 20 38

https://user-images.githubusercontent.com/48885911/189160602-c69b3cb1-ba43-47a1-ae72-67ce079474cb.mov

### CODE REVIEW

**Comment by alejomendoza** [comment]
Screen Shot 2022-09-08 at 3 48 36 PM
Not sure why I got into this state, but I couldn't click mint, seems disabled for some reason. Looking good so far!

**Comment by alejomendoza** [comment]
ok cool, I know what happened. 

- Artists should be able to mint without setting a price at first. 

- There should be a way to replace the artwork or the media file.

- There should be a way to pick a file again right now the only way is to refresh the page.

**Comment by alejomendoza** [comment]
Screen Shot 2022-09-08 at 4 09 29 PM
getting cors errors on localhost not sure why

**Comment by alejomendoza** [comment]
Screen Shot 2022-09-08 at 4 11 27 PM
broke on indexEntry (504)

**Comment by alejomendoza** [comment]
errors are looking way better than before, thanks!

**Review by co012 — APPROVED** [review]
LGTM, maybe we should unify Edit-profile, Withdraw credits, and MintScreen TextInput looks

**Comment by jmysliv** [comment]
* I've made price and equity optional
* I've added the possibility to remove currently picked media and select a different one
* other minor fixes

When it comes to this 504 Gateway Timeout error, I've noticed that on staging it occurs randomly. I tried running the server locally, but it obviously worked then, so it's hard to find out what is taking so much time. If I have to guess, I'd say the write request to algolia is taking longer than expected.

@alejomendoza

**Comment by jmysliv** [comment]
I've checked the error logs on vercel. Do we have a timeout set anywhere in the backend?
Screenshot 2022-09-12 at 11 58 02

File: packages/app/features/dashboard/profile/textEllipsis.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ export function TextEllipsis({ text }: { text: string }) {
       }}
     >
       <Text
-        style={tw.style("text-xs font-bold absolute leading-5")}
+        style={tw.style("text-xs font-bold absolute")}
```
Comment (co012) [inline packages/app/features/dashboard/profile/textEllipsis.tsx:40]: Without leading-5 the text is off-center

**Review by alejomendoza — APPROVED** [review]

File: packages/app/ui/inputs/UploadInputWithIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  };
+
+  const UploadWidget = () => {
+    if (loading) {
+      return <ActivityIndicator size="small" color={tw.color("white")} />;
+    }
+    if (success) {
+      return (
+        <View className="flex-row justify-between flex-1 pr-4 items-center">
+          <CheckIcon size={30} color={tw.color("lightGreen")} />;
```
Comment (co012) [inline packages/app/ui/inputs/UploadInputWithIcon.tsx:78]:
```suggestion
          <CheckIcon size={30} color={tw.color("lightGreen")} />
```

**Review by co012 — APPROVED** [review]
LGTM

**Pull request merged**